### DISCUSSION TITLE
Internal Server Error - Microsoft OAuth

### DISCUSSION THREAD

**Discussion by iunknown80** [open-webui/open-webui#6464] [Issues]
Hi,

I've deployed the Open WebUI application in a Docker container and hosted it on Azure App Service. The application is configured with Microsoft OAuth to allow employees SSO access to Open WebUI. We initially had a smooth login experience and were able to use the application successfully. However, after a few days, we began encountering the following issues:

1. After clicking the SSO button, the screen displays a 'successful login' message, but the page remains blank. Inspecting with Chrome developer tools shows an error at the following endpoint: api/v1/users/user/settings. The error message is (410 Unauthorized): "Your session has expired or the token is invalid. Please sign in again." Despite the successful SSO message, this issue persists.
2. Occasionally, we receive a 501 Internal Error. Logs indicate that "Authorization code was already redeemed"
[open_webui.main] OAuth callback error: invalid_grant: AADSTS54005: OAuth2 Authorization code was already redeemed, please retry with a new valid code or use an existing refresh token. Trace ID: b87e8b61-0fa7-4e11-b55e-eff34f518100 Correlation ID: 0ec3a0a4-5f03-4618-8985-0cb5e1bf038d Timestamp: 2024-10-25 20:12:59Z

Currently, we have one admin and five test users successfully registered who were actively using the application until these issues began to arise. This is now blocking our testing. Any assistance would be greatly appreciated.

Thanks

**Comment by giggio** [comment]
@iunknown80 I came across your question when I was having the same problem in Azure Container Apps. You probably have a similar setup to mine. The problem is the cifs (SMB) mount from an Azure Storage Files. You need to add a `nobrl` option to the mount. That worked for me. I don't know how you do it in Azure App Service but there should be a way.
Check out:
https://lchenghui.com/nobrl-for-mount-cifs

Other errors I had (to help others who are searching):
- "database is locked" (from app startup log, from sqlite3)
- "Your session has expired or the token is invalid" (on the api responses)
- "Error getting function valves by id" (from the app startup log)

I hope this helps someone.
